.github/workflows/test.yml

```diff
@@ -446,10 +457,25 @@ jobs:
             gcloud compute instances delete "${INSTANCE}" --zone "${{ env.ZONE }}" --delete-disks all --quiet
           fi
 
-      - name: Get disk state name from gcloud
-        id: get-disk-name
+      # If a triggering file for a rebuild was changed on a PR or we manually triggered a rebuild,
+      # use the latest available disk state as the regenerate-stateful-disks job was executed
+      - name: Get latest state disk name from gcloud
+        id: get-latest-disk-name
+        if: ${{ (needs.build.outputs.any_changed == 'true' && github.event_name == 'pull_request') || github.event.inputs.regenerate-disks == 'true' }}
```

Comment:
Question: why is this condition different from the previous steps? It is explained below, but it was not clear to me from this comment.

Reply:
Fixed by: [docs(ci): improve reasoning behind new disk selection conditions](https://github.com/ZcashFoundation/zebra/pull/4074/commits/3d7aec7512fce67adcbe97e12082a1847ee2098e)